### PR TITLE
Add codes.nora.gDiceRoller Flatpak

### PR DESCRIPTION
**Pull request by NoraCodes**
I'm not sure that all my metadata is in the right place; will the Flatpak be aware of my appdata.xml in the data/ directory of my actual application's repository?

Otherwise, I followed all instructions.

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build codes.nora.gDiceRoller

**Comment by NoraCodes** [comment]
`Can't specify build-args in sandboxed build`

This is problematic, as it's necessary for Cargo to use the network.

**Comment by nedrichards** [comment]
@NoraCodes you can't use the network with cargo on flathub, we need all the source code reproducible and archived. However, there are some helpful tools that make that easy!

https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo

could well help! there are also some other examples of rust apps on flathub that use it, have a quick search through the github repo.

**Comment by NoraCodes** [comment]
Understood, thank you @nedrichards . Should be working now.

**Comment by bilelmoussaoui** [comment]
bot, build codes.nora.gDiceRoller

File: codes.nora.gDiceRoller.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "app-id" : "codes.nora.gDiceRoller",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.32",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "codes.nora.gDiceRoller",
+    "tags" : [],
```
Comment (bilelmoussaoui) [inline codes.nora.gDiceRoller.json]:
```suggestion
```

File: codes.nora.gDiceRoller.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "desktop-file-name-suffix" : "",
```
Comment (bilelmoussaoui) [inline codes.nora.gDiceRoller.json]:
```suggestion
```

File: codes.nora.gDiceRoller.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "finish-args" : [
+        "--socket=x11",
+        "--socket=wayland",
+        "--device=dri"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [],
```
Comment (bilelmoussaoui) [inline codes.nora.gDiceRoller.json]:
```suggestion
```

**Comment by NoraCodes** [comment]
> `stat: cannot stat 'builddir/*/share/app-info/xmls/codes.nora.gDiceRoller.xml.gz': No such file or directory`

Ah, this answers my question. This should definitely go in the Flatpak documentation under Desktop Integration, I was wondering about this. Will fix momentarily. Can I ask the bot to run once I've done that, or is that privileged access?

**Comment by NoraCodes** [comment]
bot, build codes.nora.gDiceRoller

**Comment by NoraCodes** [comment]
bot, build codes.nora.gDiceRoller

**Comment by nedrichards** [comment]
@NoraCodes you should put the appdata file into /app/share/metainfo/codes.nora.gDiceRoller.appdata.xml it's then compiled by the build process along with the desktop file and icons to make that file you were looking at. Good point about clearer documentation though.

File: codes.nora.gDiceRoller.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    "app-id" : "codes.nora.gDiceRoller",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.32",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "codes.nora.gDiceRoller",
+    "finish-args" : [
+        "--socket=x11",
```
Comment (nedrichards) [inline codes.nora.gDiceRoller.json]: you should also add --share=ipc per http://docs.flatpak.org/en/latest/sandbox-permissions.html

Comment (NoraCodes) [reply codes.nora.gDiceRoller.json]: Will do, thanks so much for your help!

**Comment by NoraCodes** [comment]
bot, build codes.nora.gDiceRoller

**Comment by NoraCodes** [comment]
bot, build codes.nora.gDiceRoller

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/codes.nora.gDiceRoller

**Pull request closed without merging**